staging/src/k8s.io/legacy-cloud-providers/aws/aws_loadbalancer.go

```diff
@@ -1239,14 +1247,17 @@ func elbListenersAreEqual(actual, expected *elb.Listener) bool {
 	return true
 }
 
-func createSubnetMappings(subnetIDs []string, allocationIDs []string) []*elbv2.SubnetMapping {
+func createSubnetMappings(subnetIDs []string, allocationIDs []string, privateIPs []string) []*elbv2.SubnetMapping {
 	response := []*elbv2.SubnetMapping{}
 
 	for index, id := range subnetIDs {
 		sm := &elbv2.SubnetMapping{SubnetId: aws.String(id)}
 		if len(allocationIDs) > 0 {
 			sm.AllocationId = aws.String(allocationIDs[index])
 		}
+		if len(privateIPs) > 0 {
+			sm.PrivateIPv4Address = aws.String(privateIPs[index])
```

Comment:
Unlike EIP, the private IP has to be from the subnet range. So this will might not work in all cases.

Reply:
i think we can validate each subnet's CIDR have an privateIP assigned :D

Reply:
The subnets are auto-discovered and are sorted order by the AZ names. So it is possible to get a working configuration by specifying the private IPs in the correct order.

Validating against the CIDR range is definitely better and would lead to less configuration issues for the end user.